browser/brave_content_browser_client.cc

```diff
@@ -454,18 +453,28 @@ BraveContentBrowserClient::CreateURLLoaderThrottles(
       request, browser_context, wc_getter, navigation_ui_data,
       frame_tree_node_id);
 #if BUILDFLAG(ENABLE_SPEEDREADER)
+  using DistillState = speedreader::SpeedreaderTabHelper::DistillState;
   content::WebContents* contents = wc_getter.Run();
   if (!contents) {
     return result;
   }
   auto* tab_helper =
       speedreader::SpeedreaderTabHelper::FromWebContents(contents);
-  if (tab_helper && tab_helper->IsActiveForMainFrame() &&
+  if (!tab_helper)
+    return result;
+  const auto state = tab_helper->PageDistillState();
+  if (state != DistillState::kNone &&
```

Comment:
it seems it is the same as checking `IsActiveForMainFrame()` ?

Reply:
changed